test/ui/test_ui_login.py

```
@@ -0,0 +1,246 @@
+    """Get login password from config file or environment
+
+    Returns the plaintext password that should be used for login.
+    For test/dev environments, tries common test passwords and defaults.
+    Returns None if password cannot be determined (will skip test).
+    """
+    # Try environment variable first (for testing)
+    if os.getenv("LOGIN_PASSWORD"):
+        return os.getenv("LOGIN_PASSWORD")
+
+    # SHA256 hash of "password" - the default test password (from index.php)
+    DEFAULT_PASSWORD_HASH = '8d969eef6ecad3c29a3a629280e686cf0c3f5d5a86aff3ca12020c923adc6c92'
```

Ruff (0.15.1)

[error] Possible hardcoded password assigned to: "DEFAULT_PASSWORD_HASH"

(S105)